client/init.c

```diff
@@ -155,21 +155,26 @@ int xdag_init(int argc, char **argv, int isGui)
 		} else if(ARG_EQUAL(argv[i], "", "-threads")) { /* number of transport layer threads */
 			if (!(++i < argc && sscanf(argv[i], "%d", &transport_threads) == 1))
 				printf("Number of transport threads is not given.\n");
-		} else if(ARG_EQUAL(argv[i], "", "-dm")) {
+		} else if(ARG_EQUAL(argv[i], "", "-dm")) { /* disable mining */
 			g_disable_mining = 1;
-		} else if(ARG_EQUAL(argv[i], "", "-tag")) {
+		} else if(ARG_EQUAL(argv[i], "", "-tag")) { /* pool tag */
 			if(i+1 < argc) {
-				if(strlen(argv[i+1]) < 32 && validate_ascii(argv[i+1])) {
-					strcpy(g_pool_tag, argv[i+1]);
+				if(validate_remark(argv[i+1])) {
+					char pool_tag_buf[33] = {0};
+					strncpy(pool_tag_buf, argv[i+1], sizeof(xdag_remark_t));
```

Comment:
Why do you need intermediate copying to temporary buffer?

Reply:
This line is to protect g_pool_tag from memory issue, since the length of argv[i+1] unpredictable, so I use strncpy here to secure the copy then memcpy to g_pool_tag.

Reply:
But you can do strncpy directly to g_pool_tag. Isn't it?

Reply:
I think g_pool_tag is not char [], it's uint8 []. Let me check strncpy.